test/api_test.js

```diff
@@ -569,6 +622,105 @@ describe('Api', () => {
     });
   });
 
+  describe('credentials', () => {
+    const P = f => path.resolve(__dirname, f);
+
+    const loadFile = (filename) => {
+      return fs.readFileSync(P(filename)).toString('base64');
+    };
+
+    const captureError = async(fn, args) => {
+      try {
+        await fn.apply(client, args);
+        return 'Should fail';
+      } catch (err) {
+        return err.code;
+      }
+    };
+
+    const expectedError = 'AuthorizationFailed';
+    const createTemporaryCredentials = taskcluster.createTemporaryCredentials;
+
+    before(() => {
+      taskcluster.createTemporaryCredentials = sandbox.stub();
```

Comment:
The reason why we're using the sinon sandbox is to avoid having to do these special cleanup-requiring things.  Please use the `sandbox.stub(.....)` call.  The overall `before()` and `after()` functions should already take care of the cleanup in the `after()` here.